app/controllers/lab_tests_controller.rb

```diff
@@ -33,17 +33,25 @@ def edit
 
   # POST /lab_tests or /lab_tests.json
   def create
+    @lab_test = current_user.lab_tests.build(lab_test_params)
     authorize @lab_test
 
-    respond_to do |format|
-      if @lab_test.save
-        format.html { redirect_to @lab_test, notice: t('.success') }
-        format.json { render :show, status: :created, location: @lab_test }
+    ActiveRecord::Base.transaction do
+      @health_record = HealthRecord.new(
+        user: determine_user,
+        notes: lab_test_params[:notes]
+      )
+
+      @lab_test.recordable = @health_record
```

Comment:
Have you tried do this with a relation while you are already building a LabTest model, like I said in my [previous comment](https://github.com/Code-the-Dream-School/health-keeper/pull/23#discussion_r1887950605) about this line?